headers/complex_h.md

```diff
@@ -1,8 +1,229 @@
 # \<complex.h\>
 
-Reference: [\<complex.h\>](https://en.cppreference.com/w/c/complex)
+C Reference: [\<complex.h\>][0]
 
-## Types
+Relevant Zig documentation:
+- [std.math.complex][20]
 
-## Functions
+## [C imaginary Type][1]
 
+This type allows you to write imaginary constants in C:
+
+```c
+double imaginary i = 2.0 * I; // pure imaginary
+double complex z = 1.0 + i; // complex number
+```
+
+In Zig, there are no imaginary types and you will be representing
+the same using [Complex](#zig-complex-type) struct:
+
+```zig
+const complex = @import("std").math.complex;
+
+const imaginary i = 2.0;
+const z = complex.Complex { re: 1.0, im: i };
+```
+
+!>  Note that imaginary types in C don't have Zig equivalent, you have to denote them as
+a [Complex](#zig-complex-type) type.
+
+
+## [Zig Complex Type][21]
+
+In Zig complex numbers are represented as `Complex` type.
+It takes in two variables, `re` and `im` which represent real and imaginary
+numbers respectively.
+
+```zig
+const std = @import("std");
+const complex = std.math.complex;
+
+const Complex = complex.Complex;
+
+const z1 = Complex { re: 0, im: 1.0 } ;
+const z2 = Complex { re: 0, im: 2.0 };
+
+test "complex multiplication" {
+  const zSquared: Complex = complex.mul(z1, z2);
```

Comment:
we shouldn't need to declare the zSquared type here

Reply:
oh. type inference. cool. will remove this!